src/app/portlets/content-types/main/content-types.component.ts

```diff
@@ -24,133 +26,107 @@ import { MessageService } from '../../../api/services/messages-service';
-    ) {
-        super(
-            [
-                'contenttypes.fieldname.structure.name',
-                'contenttypes.content.variable',
-                'contenttypes.form.label.description',
-                'contenttypes.fieldname.entries',
-                'message.structure.delete.structure.and.content',
-                'message.structure.cantdelete',
-                'contenttypes.content.file',
-                'contenttypes.content.content',
-                'contenttypes.content.persona',
-                'contenttypes.content.widget',
-                'contenttypes.content.page',
-                'contenttypes.confirm.message.delete',
-                'contenttypes.confirm.message.delete.content',
-                'contenttypes.confirm.message.delete.warning',
-                'contenttypes.action.delete',
-                'contenttypes.action.cancel',
-                'Content-Type'
-            ],
-            messageService
-        );
-    }
+        private dotConfirmationService: DotConfirmationService,
+        private dotContentletService: DotContentletService
+    ) {}
 
-    /**
-     * Callback call from BaseComponent when the messages are received.
-     * @memberOf ContentTypesPortletComponent
-     */
-    onMessage(): void {
-        this.actionHeaderOptions = {
-            primary: {
-                command: $event => {
-                    this.createContentType($event);
-                },
-                model: [
-                    {
-                        command: $event => {
-                            this.createContentType('content', $event);
-                        },
-                        icon: 'fa-newspaper-o',
-                        label: this.i18nMessages['contenttypes.content.content']
-                    },
-                    {
-                        command: $event => {
-                            this.createContentType('widget', $event);
-                        },
-                        icon: 'fa-cog',
-                        label: this.i18nMessages['contenttypes.content.widget']
+    ngOnInit() {
+        Observable.forkJoin(
+            this.messageService.getMessages(this.i18nKeys),
+            this.dotContentletService.getContentTypes()
+        ).subscribe(val => {
+            this.i18nMessages = val[0];
+            this.actionHeaderOptions = {
+                primary: {
+                    command: $event => {
+                        this.createContentType($event);
                     },
-                    {
-                        command: $event => {
-                            this.createContentType('file', $event);
-                        },
-                        icon: 'fa-file-o',
-                        label: this.i18nMessages['contenttypes.content.file']
+                    model: []
+                }
+            };
+            val[1].filter(r => ['RECENT_CONTENT', 'RECENT_WIDGET'].indexOf(r.name) === -1).forEach(obj => {
```

Comment:
Extract these functions to external method to make this more readable.

Also, you can use ```.map``` instead of ```.forEach``` and don't do ```.push``` but just assign the whole thing becahse ```.map``` return an array.